cpp/src/arrow/flight/types.cc

```diff
@@ -463,6 +463,300 @@ arrow::Result<CancelFlightInfoRequest> CancelFlightInfoRequest::Deserialize(
   return out;
 }
 
+// Helper for stringifying maps containing various types
+template <typename T>
+ostream& operator<<(std::map<std::string, T>) {
+  std::stringstream ss;
+
+  ss << '{';
+  std::string sep = "";
+  for (const auto& [k, v] : session_options) {
+    std::cout << sep << '[' << k << "]: '" << v << "', ";  // PHOXME v.ToString() not implemented yet
+    sep = ", ";
+  }
+  ss << '}';
+
+  return ss.str();
+}
+static bool CompareSessionOptionMaps(
```

Comment:
Helpers can go into the anonymous namespace instead of a global static symbol

Reply:
Ty, done.